File: javascript/problem017.js
function one(n){
  if(n==1){
    return "one";
  }
  else if(n==2){
    return "two";
  }
  else if(n==3){
    return "three";
  }
  else if(n==4){
    return "four";
  }
  else if(n==5){
    return "five";
  }
  else if(n==6){
    return "six";
  }
  else if(n==7){
    return "seven";
  }
  else if(n==8){
    return "eight";
  }
  else if(n==9){
    return "nine";
  }
}

function ten(n){
  if(n==10){
    return "ten";
  }
  else if(n==11){
    return "eleven";
  }
  else if(n==12){
    return "twelve";
  }
  else if(n==13){
    return "thirteen";
  }
  else if(n==14){
    return "fourteen";
  }
  else if(n==15){
    return "fifteen";
  }
  else if(n==16){
    return "sixteen";
  }
  else if(n==17){
    return "seventeen";
  }
  else if(n==18){
    return "eighteen";
  }
  else if(n==19){
    return "nineteen";
  }
}

function ty(n){
  if(n==2){
    return "twenty";
  }
  else if(n==3){
    return "thirty";
  }
  else if(n==4){
    return "forty";
  }
  else if(n==5){
    return "fifty";
  }
  else if(n==6){
    return "sixty";
  }
  else if(n==7){
    return "seventy";
  }
  else if(n==8){
    return "eighty";
  }
  else if(n==9){
    return "ninety";
  }
}

function toStr(n){
  var returnStr;
  if(n >= 1 && n <= 9){
    returnStr = one(n);
  }
  else if(n >= 10 && n <= 19){
    returnStr = ten(n % 100)
  }
  else if(n >= 20 && n <= 99){
    if(n % 10 == 0){
      returnStr = ty(parseInt(parseInt(n % 100) / 10));
    }
    else{
      returnStr = ty(parseInt(parseInt(n % 100) / 10))+one(parseInt(parseInt(n % 100) % 10));
    }
  }
  return returnStr;
}

var result = 0;
for(i=1;i<=1000;i++){
  var str = "";
  var number = i;
  if(number >= 1 && number < 100){
    str = str + toStr(number);
  }
  else if(number >= 100 && number < 1000){
    str = one(parseInt(number / 100))+"hundred";
    number = number - parseInt(number / 100) * 100;
    if(number != 0){
      str = str + "and" + toStr(number);
    }
  }
  else if(number == 1000){
      str ="onethousand";
  }
  result += str.length;
}

console.log(result);

File: python/problem017.py
def one(n):
    if n == 1:
        return "one"
    elif n == 2:
        return "two"
    elif n == 3:
        return "three"
    elif n == 4:
        return "four"
    elif n == 5:
        return "five"
    elif n == 6:
        return "six"
    elif n == 7:
        return "seven"
    elif n == 8:
        return "eight"
    elif n == 9:
        return "nine"


def ten(n):
    if n == 10:
        return "one"
    elif n == 11:
        return "eleven"
    elif n == 12:
        return "twelve"
    elif n == 13:
        return "thirteen"
    elif n == 14:
        return "fourteen"
    elif n == 15:
        return "fifteen"
    elif n == 16:
        return "sixteen"
    elif n == 17:
        return "seventeen"
    elif n == 18:
        return "eighteen"
    elif n == 19:
        return "nineteen"


def ty(n):
    if n == 2:
        return "twenty"
    elif n == 3:
        return "thirty"
    elif n == 4:
        return "forty"
    elif n == 5:
        return "fifty"
    elif n == 6:
        return "sixty"
    elif n == 7:
        return "seventy"
    elif n == 8:
        return "eighty"
    elif n == 9:
        return "ninety"


def to_string(n):
    return_str = ""
    if 1 <= n <= 9:
        return_str = one(n)
    elif 10 <= n <= 19:
        return_str = ten(n % 100)
    elif 20 <= n <= 99:
        if n % 10 == 0:
            return_str = ty(int(int(n % 100) / 10))
        else:
            return_str = ty(int(int(n % 100) / 10)) + one(int(int(n % 100) % 10))
    return return_str


result = 0
for i in range(1, 1001):
    num_str = ""
    number = i
    if 1 <= number < 100:
        num_str += to_string(number)
    elif 100 <= number < 1000:
        num_str = one(int(number / 100)) + "hundred"
        number -= int(number / 100) * 100
        if number != 0:
            num_str += "and" + to_string(number)
    elif number == 1000:
        num_str = "onethousand"
    result += len(num_str)

print(result)